### PR TITLE
[Hosting] Remove obsolete methods ahead of 1.4.0 release

### PR DESCRIPTION
**Pull request by CodeBlanch**
**If you are reading this because you upgraded to 1.4.0 and your code no longer compiles, sorry! Much discussion went into this and we decided as a group it was the best path forward to not have obsolete code in our first shipped stable version. If you want to get up and running using the old API, you may use the RC version: [1.4.0-rc.4](https://www.nuget.org/packages/OpenTelemetry.Extensions.Hosting/1.4.0-rc.4). That version contains everything in 1.4.0 minus this PR removing the old API.**

## Migration steps

Old style:

```csharp
services.AddOpenTelemetryTracing(builder => /* configuration */);
services.AddOpenTelemetryMetrics(builder => /* configuration */);
```

New style:

```csharp
services.AddOpenTelemetry()
  .WithTracing(builder => /* configuration */)
  .WithMetrics(builder => /* configuration */);
```

Other APIs...

* `Configure` is available by casting the provider builder as either `IDeferredTracerProviderBuilder` or `IDeferredMeterProviderBuilder`.
* `GetServices` is now available as `ConfigureServices`.

## Changes

* Removes all the hosting APIs obsoleted in v1.4.0 development.

## TODOs

* [X] Appropriate `CHANGELOG.md` updated for non-trivial changes
* [ ] Changes in public API reviewed

### CODE REVIEW

File: src/OpenTelemetry.Extensions.DependencyInjection/README.md [hunk 1]
```diff
@@ -6,7 +6,7 @@
 ## Installation
 
 ```shell
-dotnet add package --prerelease OpenTelemetry.Extensions.DependencyInjection
+dotnet add package OpenTelemetry.Extensions.DependencyInjection
```
Comment (alanwest) [inline src/OpenTelemetry.Extensions.DependencyInjection/README.md:9]: 🎉

File: src/OpenTelemetry.Extensions.Hosting/OpenTelemetry.Extensions.Hosting.csproj [hunk 2]
```diff
@@ -1,13 +1,12 @@
...
+
   <PropertyGroup>
     <!-- OmniSharp/VS Code requires TargetFrameworks to be in descending order for IntelliSense and analysis. -->
     <TargetFrameworks>netstandard2.0</TargetFrameworks>
     <Description>Contains extensions to start OpenTelemetry in applications using Microsoft.Extensions.Hosting</Description>
     <RootNamespace>OpenTelemetry</RootNamespace>
     <MinVerTagPrefix>core-</MinVerTagPrefix>
-
-    <!-- this is temporary. will remove in future PR. -->
-    <Nullable>disable</Nullable>
```
Comment (TimothyMothra) [inline src/OpenTelemetry.Extensions.Hosting/OpenTelemetry.Extensions.Hosting.csproj:10]: 🎉

Comment (CodeBlanch) [reply src/OpenTelemetry.Extensions.Hosting/OpenTelemetry.Extensions.Hosting.csproj:10]: There's so little in this project now I figured might as well get it all updated while I was at it 😄

**Comment by TimothyMothra** [comment]
bump to remove stale

**Review by alanwest — APPROVED** [review]
🎉 LGTM.

**Review by utpilla — APPROVED** [review]

File: src/OpenTelemetry.Api.ProviderBuilderExtensions/README.md [hunk 3]
```diff
@@ -6,7 +6,7 @@
 ## Installation
 
 ```shell
-dotnet add package --prerelease OpenTelemetry.Api.ProviderBuilderExtensions
+dotnet add package OpenTelemetry.Api.ProviderBuilderExtensions
```
Comment (cijothomas) [inline src/OpenTelemetry.Api.ProviderBuilderExtensions/README.md:9]: lets keep this separate. The changelog points to this PR, so someone following it should be shown only whats really changed. (These are just noise for them)

File: src/OpenTelemetry.Extensions.Hosting/CHANGELOG.md [hunk 4]
```diff
@@ -11,6 +11,10 @@ Released 2023-Feb-10
   traces and/or metrics.
   ([#4174](https://github.com/open-telemetry/opentelemetry-dotnet/pull/4174))
 
+* Removed deprecated extensions: `AddOpenTelemetryTracing`,
```
Comment (Kielek) [inline src/OpenTelemetry.Extensions.Hosting/CHANGELOG.md:14]: Unreleased section?

Comment (CodeBlanch) [reply src/OpenTelemetry.Extensions.Hosting/CHANGELOG.md]: Good catch! Not sure what happened there, probably a merge thing.

**Review by cijothomas — APPROVED** [review]

**Pull request merged**